### PR TITLE
Add black and pylint  

### PR DESCRIPTION
**Pull request by Herklos**
Depends on https://github.com/Drakkar-Software/OctoBot-Commons/pull/56

Should be merged with :
https://github.com/Drakkar-Software/OctoBot-Notifications/pull/15
https://github.com/Drakkar-Software/OctoBot/pull/1120
https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/185
https://github.com/Drakkar-Software/OctoBot-Interfaces/pull/32

### CODE REVIEW

File: octobot_commons/logging/logging_util.pxd [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+cpdef void set_global_logger_level(level)
+cpdef object get_global_logger_level()
+cpdef object get_logger(str logger_name=*)
+cpdef void set_logging_level(list logger_names, object level)
+
+cdef class BotLogger:
```
Comment (GuillaumeDSM) [inline octobot_commons/logging/logging_util.pxd:23]: 💯

File: octobot_commons/pretty_printer.pxd [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# cython: language_level=3
```
Comment (GuillaumeDSM) [inline octobot_commons/pretty_printer.pxd:1]: 👍

File: octobot_commons/tentacles_management/advanced_manager.py [hunk 3]
```diff
@@ -200,23 +249,41 @@ def get_all_classes_from_parent(parent_class) -> list:
...
+    Search for a class name in a class list
+    :param class_name: the class name to search
+    :param parent_class_list: the class list
+    :return: the class if found else None
+    """
     for subclass in parent_class_list:
         if isinstance(subclass, list):
             return search_class_name_in_class_list(class_name, subclass)
-        elif subclass.get_name() == class_name:
+        if subclass.get_name() == class_name:
```
Comment (GuillaumeDSM) [inline octobot_commons/tentacles_management/advanced_manager.py:261]: 👍

File: octobot_commons/tentacles_management/advanced_manager.py [hunk 4]
```diff
@@ -200,23 +249,41 @@ def get_all_classes_from_parent(parent_class) -> list:
...
+    :return: the single deepest child class
+    """
     children_classes = clazz.__subclasses__()
     if len(children_classes) == 0:
         return clazz
     if len(children_classes) > 1:
-        get_logger(__name__).error(f"More than one child class of {clazz}, expecting one, "
-                                   f"using {children_classes[0]}")
+        get_logger(__name__).error(
+            f"More than one child class of {clazz}, expecting one, "
```
Comment (GuillaumeDSM) [inline octobot_commons/tentacles_management/advanced_manager.py:286]: might be better on one line

File: octobot_commons/tentacles_management/class_inspector.py [hunk 5]
```diff
@@ -18,61 +18,125 @@
 
 
 def default_parent_inspection(element, parent):
+    """
+    Check if the element base has the specified parent
```
Comment (GuillaumeDSM) [inline octobot_commons/tentacles_management/class_inspector.py:22]: Check if the element base**s** has the specified parent

File: octobot_commons/tentacles_management/class_inspector.py [hunk 6]
```diff
@@ -18,61 +18,125 @@
...
+    :param parent: the expected parent
+    :return: the check result
+    """
     return parent in element.mro()
 
 
 def evaluator_parent_inspection(element, parent):
-    return hasattr(element, "get_parent_evaluator_classes") and element.get_parent_evaluator_classes(parent)
+    """
+    Check if the evaluator class has the specified parent
```
Comment (GuillaumeDSM) [inline octobot_commons/tentacles_management/class_inspector.py:42]: Recursively check if the evaluator class has the specified parent

File: octobot_commons/time_frame_manager.py [hunk 7]
```diff
@@ -27,58 +27,90 @@ def __sort_time_frames(time_frames, reverse=False):
...
+    :param reverse: if the sort should be reversed
+    :return: the time frame list sorted
+    """
     return __sort_time_frames(time_frames, reverse)
 
 
-def sort_config_time_frames(config):
+def sort_config_time_frames(config) -> None:
+    """
+    Sort the time frame config and save it
```
Comment (GuillaumeDSM) [inline octobot_commons/time_frame_manager.py:52]: Sort the time frame config and save it in config

File: octobot_commons/config_manager.py [hunk 8]
```diff
@@ -50,52 +68,92 @@ def save_config(config_file, config, temp_restore_config_file, json_data=None):
...
+                    key: "" for key in CONFIG_EXCHANGE_ENCRYPTED_VALUES
+                }
 
         return dump_json(config)
     except ImportError:
-        get_logger().error(f"OctoBot_Commons/config_manager.py/jsonify_config requires "
-                           f"OctoBot-Trading package installed")
+        get_logger().error(
+            "OctoBot_Commons/config_manager.py/jsonify_config requires "
+            "OctoBot-Trading package installed"
```
[the remark this answers is not in the document]
Comment (Herklos) [reply octobot_commons/config_manager.py:145]: Yes, instead the line was too long

File: octobot_commons/tentacles_management/advanced_manager.py (hunk 4 above)
Comment (Herklos) [reply octobot_commons/tentacles_management/advanced_manager.py:286]: It's the black formating

Comment (Herklos) [reply octobot_commons/tentacles_management/advanced_manager.py:286]: https://github.com/psf/black

**Pull request merged**